src/pynwb/image.py

```diff
@@ -73,19 +79,104 @@ def __init__(self, **kwargs):
         if unit is None:
             kwargs['unit'] = ImageSeries.DEFAULT_UNIT
 
+        if (
+            args_to_set["external_file"] is not None
+            and args_to_set["starting_frame"] is None
+        ):
+            args_to_set["starting_frame"] = (
+                [0] if len(args_to_set["external_file"]) == 1 else None
+            )
+
         super().__init__(**kwargs)
 
-        if args_to_set["external_file"] is None:
-            args_to_set["starting_frame"] = None  # overwrite starting_frame
         for key, val in args_to_set.items():
             setattr(self, key, val)
 
+        if not self._check_external_file_starting_frame_length():
+            msg = ("%s '%s': The number of frame indices in 'starting_frame' should have "
+                   "the same length as 'external_file'." % (self.__class__.__name__, name))
+            self._raise_error_when_check_is_violated(error_msg=msg)
+
+        if self._change_external_file_format():
+            warnings.warn(
+                "%s '%s': The value for 'format' has been changed to 'external'. "
+                "Setting a default value for 'format' is deprecated and will be changed "
+                "to raising a ValueError in the next release."
+                % (self.__class__.__name__, self.name),
+                DeprecationWarning,
+            )
+
+        if not self._check_external_file_format():
+            msg = (
+                "%s '%s': Format must be 'external' when external_file is specified."
+                % (self.__class__.__name__, name)
+            )
+            self._raise_error_when_check_is_violated(error_msg=msg)
+
+        if not self._check_external_file_data():
+            msg = (
+                "%s '%s': Either external_file or data must be specified (not None), but not both."
+                % (self.__class__.__name__, name)
+            )
+            self._raise_error_when_check_is_violated(error_msg=msg)
+
```

Comment:
This pattern feels a bit verbose, in that it requires a lot of if statements with checks, generation of messages and calls to raise. To make ``__init__`` more compact and to have the checks more self-contained, I think one option would be to:

1. Have the ``_check`` functions return the message string if an error should be raised and return None if the check is passed
2. In `` self._raise_error_when_check_is_violated(error_msg=msg)`` we can then check if ``error_msg is None ``and simply return if it is ``None`` (I added a corresponding suggestion below)

With these changes the code above would simplify to:

```
self._raise_error_when_check_is_violated(error_msg=self._check_external_file_starting_frame_length())
self._raise_error_when_check_is_violated(error_msg=self._check_external_file_format())
self._raise_error_when_check_is_violated(error_msg=self._check_external_file_data())
```

Or one could in this way even write this a loop, but I think that only makes sense when there are a very large number of checks.

```
data_checks = [self._check_external_file_starting_frame_length, self._check_external_file_format, self._check_external_file_data]
for check in data_checks:
    self._raise_error_when_check_is_violated(error_msg=check)
```